### PR TITLE
eng-432: Handle push notification with image

### PR DESCRIPTION
**Pull request by bhanuprakash1226**

### CODE REVIEW

File: Sources/DashX/DashXClient.swift [hunk 1]
```diff
@@ -650,4 +650,46 @@ public class DashXClient {
...
+                                         notificationContent: UNMutableNotificationContent) {
+        let task = URLSession.shared.downloadTask(with: imageURL) { (location, response, error) in
+            guard let location = location else {
+                return
+            }
+
+            // Create a temporary file URL to save the downloaded image
+            let tmpDirectoryURL = FileManager.default.temporaryDirectory
+            let uuid = UUID().uuidString
+            let tmpFileURL = tmpDirectoryURL.appendingPathComponent(uuid+".png")
```
Comment (Amandeepsinghghai) [inline Sources/DashX/DashXClient.swift]:
```suggestion
            let tmpFileURL = tmpDirectoryURL.appendingPathComponent(uuid + ".png")
```

File: Sources/DashX/DashXClient.swift [hunk 2]
```diff
@@ -650,4 +650,46 @@ public class DashXClient {
...
     }
+    
+    // MARK: - Create Notification
+    
+    func createNotification(id: String, content: UNMutableNotificationContent) {
+        let request = UNNotificationRequest(identifier: id, content: content, trigger: nil)
+
+        UNUserNotificationCenter.current().add(request) { error in
+            if let error = error {
+                print("Error while adding notification request: \(error.localizedDescription)")
```
Comment (Amandeepsinghghai) [inline Sources/DashX/DashXClient.swift]:
```suggestion
                DashXLog.d(tag: #function, "Error while adding notification request: \(error.localizedDescription)")
```

File: Sources/DashX/DashXClient.swift [hunk 3]
```diff
@@ -650,4 +650,46 @@ public class DashXClient {
...
+            } else {
+                print("Notification request added successfully.")
```
Comment (Amandeepsinghghai) [inline Sources/DashX/DashXClient.swift]:
```suggestion
                DashXLog.d(tag: #function, "Notification request added successfully.")
```

File: Sources/DashX/DashXClient.swift [hunk 4]
```diff
@@ -650,4 +650,46 @@ public class DashXClient {
...
+            do {
+                // Move the downloaded file to the temporary file URL
+                try FileManager.default.moveItem(at: location, to: tmpFileURL)
+
+                // Create the notification attachment from the temporary file URL
+                let attachment = try UNNotificationAttachment(identifier: "image", url: tmpFileURL, options: nil)
+                notificationContent.attachments = [attachment]
+                self.createNotification(id: id, content: notificationContent)
+            } catch {
+                print("Error moving file: \(error.localizedDescription)")
```
Comment (Amandeepsinghghai) [inline Sources/DashX/DashXClient.swift]:
```suggestion
                DashXLog.d(tag: #function, "Error moving file: \(error.localizedDescription)")
```

File: Sources/DashX/DashXClient.swift [hunk 5]
```diff
@@ -650,4 +650,46 @@ public class DashXClient {
...
+            let tmpDirectoryURL = FileManager.default.temporaryDirectory
+            let uuid = UUID().uuidString
+            let tmpFileURL = tmpDirectoryURL.appendingPathComponent(uuid+".png")
+
+            do {
+                // Move the downloaded file to the temporary file URL
+                try FileManager.default.moveItem(at: location, to: tmpFileURL)
+
+                // Create the notification attachment from the temporary file URL
+                let attachment = try UNNotificationAttachment(identifier: "image", url: tmpFileURL, options: nil)
```
Comment (Amandeepsinghghai) [inline Sources/DashX/DashXClient.swift]: identifier is supposed to be a unique value. Set it to a unique value maybe something like `"\(id)-attachment"`

File: Sources/DashX/DashXClient.swift [hunk 6]
```diff
@@ -650,4 +650,46 @@ public class DashXClient {
             trackNotification(id, event, ISO8601DateFormatter.timeStamp)
         }
     }
+    
+    // MARK: - Create Notification
+    
+    func createNotification(id: String, content: UNMutableNotificationContent) {
```
Comment (Amandeepsinghghai) [inline Sources/DashX/DashXClient.swift]: Why can't we create this method in Sources/DashX/DashXAppDelegate.swift. We don't want to expose this as it is

File: Sources/DashX/DashXAppDelegate.swift [hunk 7]
```diff
@@ -74,11 +75,14 @@ open class DashXAppDelegate: UIResponder, UIApplicationDelegate, UNUserNotificat
...
         notificationContent.userInfo = userInfo
-
-        let request = UNNotificationRequest(identifier: dashxData.id, content: notificationContent, trigger: nil)
-
-        UNUserNotificationCenter.current().add(request)
-
+        
+        if let imagePath = dashxData.image,
+           let imageURL = URL(string: imagePath) {
+            dashXClient.createNotificationWithImage(id: dashxData.id, imageURL: imageURL, notificationContent: notificationContent)
```
Comment (Amandeepsinghghai) [inline Sources/DashX/DashXAppDelegate.swift]: Should `notificationContent` parameter be `content`? 
Similar to `createNotification(id: dashxData.id, content: notificationContent)`

File: Sources/DashX/DashXClient.swift (hunk 1 above)
Comment (bhanuprakash1226) [reply Sources/DashX/DashXClient.swift]: Done

File: Sources/DashX/DashXClient.swift (hunk 2 above)
Comment (bhanuprakash1226) [reply Sources/DashX/DashXClient.swift]: Done

File: Sources/DashX/DashXClient.swift (hunk 3 above)
Comment (bhanuprakash1226) [reply Sources/DashX/DashXClient.swift]: Replaced with the same.

File: Sources/DashX/DashXClient.swift (hunk 4 above)
Comment (bhanuprakash1226) [reply Sources/DashX/DashXClient.swift]: Done

File: Sources/DashX/DashXClient.swift (hunk 5 above)
Comment (bhanuprakash1226) [reply Sources/DashX/DashXClient.swift]: Replaced with the id

File: Sources/DashX/DashXClient.swift (hunk 6 above)
Comment (bhanuprakash1226) [reply Sources/DashX/DashXClient.swift]: made as a private method in the extension of DashX App delegate.

File: Sources/DashX/DashXAppDelegate.swift (hunk 7 above)
Comment (bhanuprakash1226) [reply Sources/DashX/DashXAppDelegate.swift]: Done

**Pull request merged**